pkg/lambda/lambda.go

```diff
@@ -15,59 +14,65 @@ import (
 type HandlerFactory func(ctx context.Context, config cfg.Config, logger log.Logger) (interface{}, error)
 
 func Start(handlerFactory HandlerFactory, configOptions ...cfg.Option) {
-	clock.WithUseUTC(true)
-
-	logHandler := log.NewHandlerIoWriter(log.LevelInfo, []string{}, log.FormatterConsole, "", os.Stdout)
-	loggerOptions := []log.Option{
-		log.WithHandlers(logHandler),
-		log.WithContextFieldsResolver(log.ContextLoggerFieldsResolver),
-	}
+	var err error
+	var cfgPostProcessors map[string]int
+	var handlers []log.Handler
+	var lambdaHandler any
 
+	clock.WithUseUTC(true)
+	ctx := appctx.WithContainer(context.Background())
+	config := cfg.New()
 	logger := log.NewLogger()
 
-	if err := logger.Option(loggerOptions...); err != nil {
-		logger.Error("failed to apply logger options: %w", err)
-
-		os.Exit(1)
-	}
-
 	// configure and create config
 	mergedConfigOptions := append([]cfg.Option{
 		cfg.WithEnvKeyReplacer(cfg.DefaultEnvKeyReplacer),
 		cfg.WithSanitizers(cfg.TimeSanitizer),
 		cfg.WithErrorHandlers(func(msg string, args ...interface{}) {
-			logger.Error(msg, args...)
-			os.Exit(1)
+			defaultErrorHandler(msg, args...)
 		}),
 	}, configOptions...)
 
-	config := cfg.New()
-	if err := config.Option(mergedConfigOptions...); err != nil {
-		logger.Error("failed to apply config options: %w", err)
+	if err = config.Option(mergedConfigOptions...); err != nil {
+		defaultErrorHandler("failed to apply config options: %w", err)
 
-		os.Exit(1)
+		return
 	}
 
-	if cfgPostProcessors, err := cfg.ApplyPostProcessors(config); err != nil {
-		logger.Error("can not apply post processor on config: %w", err)
+	if cfgPostProcessors, err = cfg.ApplyPostProcessors(config); err != nil {
+		defaultErrorHandler("can not apply post processor on config: %w", err)
 
-		os.Exit(1)
-	} else {
-		for name, priority := range cfgPostProcessors {
-			logger.Debug("applied priority %d config post processor '%s'", priority, name)
-		}
+		return
```

Comment:
default error handler does os.Exit(1), so you don't need to return, do you?

Reply:
but if someone overwrites the default error handler with something which doesn't do an os.Exit, I still want to make sure, that it will not continue with the lambda start.